### PR TITLE
Switch to `focus-trap`

### PR DESCRIPTION
**Pull request by dc7290**
## Summary

Change the focus trap implementation to a library called `focus-trap`.
The reasons are as follows

- Focus traps have many considerations and may be expensive to maintain
  - Here are some considerations when finding focusable elements
    - https://github.com/focus-trap/tabbable#more-details
- It is safer to use the library for behavior when a modal is opened from a modal
  - https://github.com/focus-trap/focus-trap#one-at-a-time
- Various options allow for flexibility.
  - For example, if the focusable element is not in the modal, you can control the behavior in detail.

## Notes

⚠️ This PR contains disruptive changes!

- A warning is issued when there is no focusable element in the modal.
- Use `focusOptions.clickOutsideDeactivates` instead of the `closeOnOverlayClick` option.

### CODE REVIEW

File: examples/src/index.tsx [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import React from 'react';
 import { render } from 'react-dom';
-import { useModal } from '../../dist';
+import { useModal } from '../../src';
```
Comment (dc7290) [inline examples/src/index.tsx:3]: Since the gh-pages build is bundled using webpack, why not import it directly from src?

File: package.json [hunk 2]
```diff
@@ -37,7 +38,7 @@
     "ts-loader": "^8.2.0",
     "typescript": "^4.3.5",
     "webpack": "^5.50.0",
-    "webpack-cli": "^3.3.12",
+    "webpack-cli": "^4.10.0",
```
Comment (dc7290) [inline package.json:41]: The `webpack-dev-server` has been updated in dependabot, and `webpack-cli` needs to be updated as well.

File: src/useFocusTrap.tsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    focusTrap.activate();
+
+    // It looks like we need to wait a bit until document.activeElement is updated.
+    setTimeout(() => {
+      if (document.activeElement === ref.current) {
+        console.warn(`[react-hooks-use-modal]: Since there were no focusable elements in the modal, the initial focus was on the containing element.
+  WAI-ARIA 1.1 states that there should be at least one focusable element in the modal.
+  https://www.w3.org/TR/wai-aria-1.1/#dialog`);
+      }
+    }, 100);
```
Comment (dc7290) [inline src/useFocusTrap.tsx:27]: A warning is now issued when there is no focusable element in the modal element and the containing element has the focus.

File: examples/src/index.tsx (hunk 1 above)
Comment (hiro08gh) [reply examples/src/index.tsx:3]: If you can deploy with this, no problem.

Comment (dc7290) [reply examples/src/index.tsx:3]: We've been able to get the development server up and running, and create production builds, both with no problems!

**Review by hiro08gh — APPROVED** [review]

**Comment by hiro08gh** [comment]
LGTM!

**Comment by dc7290** [comment]
Thank you!

**Pull request merged**